cmd/terramate/cli/cli.go

```diff
@@ -205,52 +200,46 @@ func newCLI(
 	logLevel := parsedArgs.LogLevel
 	logFmt := parsedArgs.LogFmt
 
-	if err := configureLogging(logLevel, logFmt, stderr); err != nil {
-		return nil, err
-	}
-
-	log.Trace().
-		Str("action", "newCli()").
-		Msg("Get working directory.")
-	wd := parsedArgs.Chdir
-	if wd == "" {
-		wd, err = os.Getwd()
-		if err != nil {
-			return nil, fmt.Errorf("failed to get working directory: %w", err)
-		}
-	}
+	configureLogging(logLevel, logFmt, stderr)
 
-	logger := log.With().
-		Str("action", "newCli()").
-		Str("stack", wd).
-		Logger()
+	logger := log.With().Logger()
 
-	logger.Trace().
-		Msgf("Evaluate symbolic links for %q.", wd)
-	wd, err = filepath.EvalSymlinks(wd)
+	cwd, err := os.Getwd()
 	if err != nil {
-		return nil, fmt.Errorf("failed evaluating symlinks for %q: %w", wd, err)
+		log.Fatal().
+			Err(err).
+			Msg("Getwd() failed")
 	}
 
-	logger.Trace().
-		Msgf("Get absolute file path of %q.", wd)
-	wd, err = filepath.Abs(wd)
-	if err != nil {
-		return nil, fmt.Errorf("getting absolute path of %q: %w", wd, err)
-	}
+	if parsedArgs.Chdir != "" {
+		logger.Debug().
+			Str("cwd", cwd).
+			Str("dir", parsedArgs.Chdir).
+			Msg("Changing working directory")
+		err = os.Chdir(parsedArgs.Chdir)
```

Comment:
it was a previous design decision to test the cli logic as a function, so we avoid calling anything that changes the process working dir, since that changes the overall test process environment, so things breaking are probably related to os.Getwd/os.Chdir calls.

One alternative discussed at the time, that would improve isolation, is to test the cli by building/running the terramate binary, then since it is run as a separate process from the test itself this would not be an issue.